docs/src/main/sphinx/connector/iceberg.md

```diff
@@ -1814,3 +1814,34 @@ The connector supports redirection from Iceberg tables to Hive tables with the
 
 The connector supports configuring and using [file system
 caching](/object-storage/file-system-cache).
+
+### In-memory Iceberg metadata caching
+
+The connector supports caching Iceberg metadata in memory. This caching is
+enabled by default, and uses the coordinator node of the cluster. To deactivate
+the caching, set the `iceberg.metadata-cache.enabled` configuration property to
+`false`.
+
+Additionally, you can use the following catalog configuration properties:
+
+:::{list-table} Metadata caching configuration properties
+:widths: 25, 75
+:header-rows: 1
+
+* - Property
+  - Description
+* - `fs.memory-cache.ttl`
+  - The maximum [duration](prop-type-duration) to keep files in the cache prior
+    to eviction. The minimum value of `0s` means that caching is effectively
+    turned off. Defaults to `1h`.
+* - `fs.memory-cache.max-size`
+  - The maximum total [data size](prop-type-data-size) of the cache. When
+    raising this value keep in mind that the coordinator memory is used.
+    Defaults to `200MB`.
+* - `fs.memory-cache.max-content-length`
+  - The maximum file size that can be cached. Defaults to `15MB`.
+:::
+
+When file system caching from `fs.cache.enabled` is enabled, Iceberg metadata is
```

Comment:
Move this up into the intro of this section and add this info first. Then talk about the in-memory stuff